lib/SPIRV/SPIRVWriter.cpp

```diff
@@ -670,7 +681,8 @@ SPIRVType *LLVMToSPIRVBase::transPointerType(Type *ET, unsigned AddrSpc) {
 
     auto SaveType = [&](SPIRVType *MappedTy) {
       OpaqueStructMap[Key] = MappedTy;
-      PointeeTypeMap[TypeKey] = MappedTy;
+      if (!IsBuiltin)
+        PointeeTypeMap[TypeKey] = MappedTy;
```

Comment:
Does it mean, that for every builtin GV translated we won't visit the map, but instead would translate a type from scratch?

Reply:
unfortunately ... yes.